server/job.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"github.com/google/uuid"
+	pb "github.com/romanlevin/felek/jobs"
+	"os/exec"
+)
+
+type job struct {
+	id        string
+	cmd       *exec.Cmd
+	stopped   bool
+	exitError string
+	owner     string
+}
+
+func jobStatus(j *job) *pb.JobStatus {
+	c := j.cmd
+	if j.exitError == "" && c.ProcessState != nil {
+		// Job exited without exitError
+		return &pb.JobStatus{
+			Id: &pb.JobID{Value: j.id},
+			JobState: &pb.JobStatus_StoppedJob{
+				StoppedJob: &pb.StoppedJob{
+					Exit: &pb.StoppedJob_ExitCode{
+						ExitCode: int64(c.ProcessState.ExitCode()),
+					},
+					SystemTime: int64(c.ProcessState.SystemTime()),
+					UserTime:   int64(c.ProcessState.UserTime()),
+					Stopped:    j.stopped,
+				},
+			},
+		}
+	} else if c.ProcessState == nil {
+		// Job still running
+		return &pb.JobStatus{
+			Id: &pb.JobID{Value: j.id},
+			JobState: &pb.JobStatus_RunningJob{
+				RunningJob: &pb.RunningJob{Pid: int64(j.cmd.Process.Pid)},
+			},
+		}
+	} else {
```

Comment:
It's pretty hard to follow the logic in this function due to verbosity and branching. I'd try to restructure it e.g. create `pb.JobStatus` once and then fill appropriate fields depending on the current job status with explanations of each case.

Reply:
@r0mant please take a look at the current implementation. I'm not super happy with it still, but Go's way of handling `oneof` fields is inherently a bit awkward.